service/service.go

```diff
@@ -6,14 +6,18 @@ import (
 	"time"
 
 	"github.com/evergreen-ci/evergreen"
+	"github.com/gorilla/csrf"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
 	"github.com/tylerb/graceful"
 )
 
 // RunGracefully borrows extensively from the grace
-func RunGracefully(addr string, timeout time.Duration, n http.Handler) error {
+func RunGracefully(addr string, timeout time.Duration, n http.Handler, csrfToken string) error {
```

Comment:
I think we should avoid changing the API of this function, and call (potentially wrapped in the util package or other function) the csrf state setup.

Reply:
Done